tools/validators/instance_validator/validate/handler.py

```diff
@@ -90,14 +93,32 @@ def RunValidation(filenames: List[str],
     print('Subscription and a service account file are '
           'both needed for the telemetry validation!')
     sys.exit(0)
-  print('\nStarting validator...\n')
-  print('\nStarting universe generation...\n')
+
+  if not filenames and not directories:
+    print('You must specify one or more file or directory paths!')
+    sys.exit(0)
+  elif filenames is None:
+    filenames = []
+
+  print('\nStarting universe generation...')
   universe = generate_universe.BuildUniverse(modified_types_filepath)
   if not universe:
     print('\nError generating universe')
     sys.exit(0)
+
   print('\nStarting config validation...\n')
+  if directories is not None and len(directories):
+    # Recursively search in the directories for YAML files, and add them to
+    # the filenames list, with directory subpath prepended.
+    for directory in directories:
+      for root, dirs, files in os.walk(directory):
+        for file in files:
+          _lowername = file.lower()
+          if _lowername.endswith('.yaml') or _lowername.endswith('.yml'):
```

Comment:
Is this officially valid for yaml?  I'd prefer to be stricter if possible.

Reply:
I looked it up and you're correct, the file extension is strongly recommended to be *.yaml , so I'll remove the *.yml check.

Reply:
@sicada any updates on this one?